Modules/_datetimemodule.c

```diff
@@ -3250,6 +3257,32 @@ date_fromtimestamp(PyObject *cls, PyObject *obj)
     if (_PyTime_ObjectToTime_t(obj, &t, _PyTime_ROUND_FLOOR) == -1)
         return NULL;
 
+#ifdef MS_WINDOWS
+    if (t < 0) {
+        if (_PyTime_localtime(0, &tm) != 0)
+            return NULL;
+
+        int normalize = 1, negate = 0;
+        PyObject *date = new_date_subclass_ex(tm.tm_year + 1900,
+                                              tm.tm_mon + 1,
+                                              tm.tm_mday,
+                                              cls);
+	if (date == NULL) {
+	    return NULL;
+	}
+	PyObject *result = NULL;
+        PyObject *delta = new_delta(0, (int)t, 0, normalize);
+	if (delta == NULL) {
+	    goto error;
+	}
+	result = add_date_timedelta(PyDate_CAST(date), PyDelta_CAST(delta), negate);
+        Py_XDECREF(delta);
+    error:
+        Py_XDECREF(date);
```

Comment:
*date* cannot be NULL:

```suggestion
        Py_DECREF(date);
```

Reply:
Done.